### PR TITLE
fix(DB/SAI): Bloodmyst Isle missing SAI

### PR DESCRIPTION
**Pull request by Knindzagxg**
Started checking lower level zones for missing SAI or other problems.

Using Kiera DB to make new, so mods, feel free to edit SQL if need.

To test:
.tele Bloodmystisle -> Check creatures around.

Apply SQL and then restart server (Or reload creature template for every ID)

### CODE REVIEW

**Comment by Pondaveia** [comment]
You need to be more specific with the creatures and what did you change so people can understand better what to test and how.

**Comment by Knindzagxg** [comment]
It's mostly missing SAI.
They had nothing, and it's updated with info from wowdb.

File: data/sql/updates/pending_db_world/rev_1556033308686302400.sql [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+-- Creating a new SmartAI script for [Creature] ENTRY 17701 (name: Lord Xiz)
+
+-- Table creature_template
+UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 17701;
+
+-- Table smart_scripts
+DELETE FROM `smart_scripts` WHERE (source_type = 0 AND entryorguid = 17701);
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(17701, 0, 0, 0, 0, 0, 100, 0, 2000, 2000, 32000, 32000, 11, 13318, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 'Lord Xiz - In Combat - Cast \'13318\'');
```
Comment (poszer) [inline data/sql/updates/pending_db_world/rev_1556033308686302400.sql]: This creature doesn't have abilities, remove it. [Creature] ENTRY 17701 (name: Lord Xiz)

File: data/sql/updates/pending_db_world/rev_1556033308686302400.sql [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+-- Creating a new SmartAI script for [Creature] ENTRY 17343 (name: Thistle Lasher)
+
+-- Table creature_template
+UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 17343;
+
+-- Table smart_scripts
+DELETE FROM `smart_scripts` WHERE (source_type = 0 AND entryorguid = 17343);
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(17343, 0, 0, 0, 0, 0, 100, 0, 1800, 1800, 8600, 8600, 11, 31286, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 'Thistle Lasher - In Combat - Cast \'31286\'');
```
Comment (poszer) [inline data/sql/updates/pending_db_world/rev_1556033308686302400.sql]: Change to [this spell](https://wotlk.evowow.com/?spell=34644)

File: data/sql/updates/pending_db_world/rev_1556033308686302400.sql [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+-- Table creature_template
+UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 17608;
+
+-- Table smart_scripts
+DELETE FROM `smart_scripts` WHERE (source_type = 0 AND entryorguid = 17608);
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(17608, 0, 0, 0, 4, 0, 100, 0, 0, 0, 0, 0, 11, 31734, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 'Sunhawk Pyromancer - On Aggro - Cast \'31734\''),
+(17608, 0, 1, 0, 0, 0, 100, 0, 1500, 1500, 25700, 25700, 11, 11962, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 'Sunhawk Pyromancer - In Combat - Cast \'11962\''),
+(17608, 0, 2, 0, 0, 0, 100, 0, 5200, 5200, 9900, 9900, 11, 9053, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 'Sunhawk Pyromancer - In Combat - Cast \'9053\'');
```
Comment (poszer) [inline data/sql/updates/pending_db_world/rev_1556033308686302400.sql]: If you can, add here one more ability [Concussive Shot](https://wotlk.evowow.com/?spell=17174)

File: data/sql/updates/pending_db_world/rev_1556033308686302400.sql [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+
+-- Table creature_template
+UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 17607;
+
+-- Table smart_scripts
+DELETE FROM `smart_scripts` WHERE (source_type = 0 AND entryorguid = 17607);
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(17607, 0, 0, 0, 4, 0, 100, 0, 0, 0, 0, 0, 11, 31734, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 'Sunhawk Defender - On Aggro - Cast \'31734\''),
+(17607, 0, 1, 0, 0, 0, 100, 0, 3200, 3200, 15400, 15400, 11, 15284, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 'Sunhawk Defender - In Combat - Cast \'15284\''),
+(17607, 0, 2, 0, 0, 0, 100, 0, 8600, 8600, 34200, 34200, 11, 31737, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 'Sunhawk Defender - In Combat - Cast \'31737\'');
```
Comment (poszer) [inline data/sql/updates/pending_db_world/rev_1556033308686302400.sql]: Here too, if you can add one more ability [Backhand](https://wotlk.evowow.com/?spell=6253)

File: data/sql/updates/pending_db_world/rev_1556033308686302400.sql [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+
+-- Creating a new SmartAI script for [Creature] ENTRY 17610 (name: Sunhawk Agent)
+
+-- Table creature_template
+UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 17610;
+
+-- Table smart_scripts
+DELETE FROM `smart_scripts` WHERE (source_type = 0 AND entryorguid = 17610);
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(17610, 0, 0, 0, 4, 0, 100, 0, 0, 0, 0, 0, 11, 31734, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 'Sunhawk Agent - On Aggro - Cast \'31734\''),
```
Comment (poszer) [inline data/sql/updates/pending_db_world/rev_1556033308686302400.sql]: Remove [31734-Mark of the Sunhawk](https://wotlk.evowow.com/?spell=31734), and add [Shadowstalker Stealth](https://wotlk.evowow.com/?spell=5916) and [Backstab](https://wotlk.evowow.com/?spell=7159) as the creature abilities

File: data/sql/updates/pending_db_world/rev_1556033308686302400.sql [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+
+-- Creating a new SmartAI script for [Creature] ENTRY 17322 (name: Infected Wildkin)
+
+-- Table creature_template
+UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 17322;
+
+-- Table smart_scripts
+DELETE FROM `smart_scripts` WHERE (source_type = 0 AND entryorguid = 17322);
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(17322, 0, 0, 0, 0, 0, 100, 0, 1500, 1500, 3200, 3200, 11, 31282, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Infected Wildkin - In Combat - Cast \'31282\'');
```
Comment (poszer) [inline data/sql/updates/pending_db_world/rev_1556033308686302400.sql]: This creature ENTRY 17322 (name: Infected Wildkin) doesn't have any abilities, so you remove it

File: data/sql/updates/pending_db_world/rev_1556033308686302400.sql [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+
+-- Creating a new SmartAI script for [Creature] ENTRY 17588 (name: Veridian Whelp)
+
+-- Table creature_template
+UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 17588;
+
+-- Table smart_scripts
+DELETE FROM `smart_scripts` WHERE (source_type = 0 AND entryorguid = 17588);
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(17588, 0, 0, 0, 0, 0, 100, 0, 7500, 7500, 28500, 28500, 11, 31292, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Veridian Whelp - In Combat - Cast \'31292\'');
```
Comment (poszer) [inline data/sql/updates/pending_db_world/rev_1556033308686302400.sql]: Same, doesn't have abilities, you can remove this also ( ENTRY 17588 (name: Veridian Whelp) )

**Comment by FrancescoBorzi** [comment]
Don't forget about http://www.azerothcore.org/wiki/Dealing-with-SQL-files#write-compact-code

**Comment by Knindzagxg** [comment]
Updated, check now again please.

**Comment by BarbzYHOOL** [comment]
Good work @Knindzagxg but please follow the issue template next time else it's not gonna be tested by anybody :(

**Comment by Knindzagxg** [comment]
Yeah, but started learning Git, u know i hate it :D

**Comment by BarbzYHOOL** [comment]
I meant Pull Request template, not issue template (you know when it asks "describe how to test the fix")

But glad you made these PRs!

**Comment by Stoabrogga** [comment]
Tested all creatures and had to adjust some of their spells / behaviour, but it should now be good to merge:

- Lord Xiz (17701) now is equipped with an appropriate weapon ("NPC Equip 14882")
- Wrathscale Screamer (17333) now uses "Scream" (31295) in combat
- Wrathscale Shorestalker (17331) now uses "Strike" (11976) in combat
- Blue Flutterer (17349) now uses "Rake" (36332) in combat
- Blacksilt Tidecaller (17327) now uses "Lightning Shield" (12550) and "Rejuvenation" (12160) in combat
- Thistle Lasher (17343) now uses "Lash" (31286) in combat (had to correct this as it was 34644 before, which would instantly kill the player; also had to link "Dropped Weapon" to it, otherwise it is not cast on the player)
- Corrupted Treant (17352) now uses "Stone Skin" (5810) in combat
- Enraged Ravager (17527) now uses "Ravage" (3242) in combat and "Enrage" (15716) when dropping under 30% health (had to fix the health check, though)
- Mutated Tangler (17346) now uses "Entangling Roots" (31287) in combat
- Corrupted Stomper (17353) now uses "Stomp" (31277) in combat
- Fouled Water Spirit (17358) now uses "Bloodmyst Chill" (31280) and "Bloodbolt" (31281) in combat
- Myst Spinner (17522) now uses "Web" (745) in combat
- Myst Leecher (17523) now uses "Leech Poison" (31288) in combat
- Sunhawk Pyromancer (17608) now uses "Immolate" (11962), "Fireball" (9053) in combat and "Mark of the Sunhawk" (31734) in combat LOS 10 yards (had to fix "Mark of the Sunhawk" and remove "Concussive Shot" as this requires a ranged weapon)
- Sunhawk Defender (17607) now uses "Cleave" (15284), "Whirlwind" (31737) in combat and "Mark of the Sunhawk" (31734) in combat LOS 10 yards (had to fix "Mark of the Sunhawk" and remove the duplicate "Whirlwind")
- Sunhawk Agent (17610) now uses "Sinister Strike" (14873), "Backstab" (7159), "Eviscerate" (15691) in combat and "Shadowstalker Stealth" (5916) on reset (had to change this from "respawn" to "reset" to recast it when leaving combat)
- Sunhawk Saboteur (17609) now uses "Shoot" (6660), "Multi-Shot" (14443) in combat and "Mark of the Sunhawk" (31734) in combat LOS 10 yards (had to fix "Mark of the Sunhawk" and the ranged combat behaviour)
- Bloodcursed Naga (17713) now uses "Frostbolt" (20792) in combat and "Frost Armor" (12544) on reset (had to change this from "respawn" to "reset" to recast it when leaving combat and also remove "Chilled" (6136) as this will be triggered by "Frost Armor")

**Comment by Stoabrogga** [comment]
Merged. Thanks @Knindzagxg for providing this PR.

**Pull request merged**